xbmc/pictures/GUIWindowSlideShow.cpp

```diff
@@ -144,6 +145,7 @@ CGUIWindowSlideShow::CGUIWindowSlideShow(void)
   m_Resolution = RES_INVALID;
   m_loadType = KEEP_IN_MEMORY;
   m_bLoadNextPic = false;
+  CServiceBroker::GetSlideShowDelegator().SetDelegate(this);
```

Comment:
Can a slideshow delegator exist without the GUI? The purpose was to make slideshow not depend on the GUI, in that the GUI can be taken down and the service won't crash. So there should be a way to `ResetDelegate()` in this window's destructor.

Reply:
Nice point considering the current implementation, will do.

If a delegate could exist without the GUI..well I guess any "external renderer" could be a slideshow delegate (but we have none). To be honest this should be modelled as just another player (a picture player) but yeah, baby steps :) My main goal here is just to decouple stuff so it can be improved incrementally